From c4ca1e102039464715fa8addc3108c5fd68896c0 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Wed, 2 Dec 2015 11:02:12 +0800
Subject: [PATCH 0755/5242] MLK-11942 ASoC: fsl_asrc_m2m: free pair after
 allocating m2m failed

commit  6fd6b599e0d8c421ed157cef8a492ade9c42c929 from
https://source.codeaurora.org/external/imx/linux-imx.git

free pair after allocating m2m failed. Reported by coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit 237a35db5b48b78250a019f891d9d746c3411d49)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_asrc_m2m.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_asrc_m2m.c b/sound/soc/fsl/fsl_asrc_m2m.c
index 8de37d4..0caed62 100644
--- a/sound/soc/fsl/fsl_asrc_m2m.c
+++ b/sound/soc/fsl/fsl_asrc_m2m.c
@@ -827,7 +827,8 @@ static int fsl_asrc_open(struct inode *inode, struct file *file)
 	m2m = kzalloc(sizeof(struct fsl_asrc_m2m), GFP_KERNEL);
 	if (!m2m) {
 		dev_err(dev, "failed to allocate m2m resource\n");
-		return -ENOMEM;
+		ret = -ENOMEM;
+		goto out;
 	}
 
 	pair->private = m2m;
@@ -843,6 +844,10 @@ static int fsl_asrc_open(struct inode *inode, struct file *file)
 	pm_runtime_get_sync(dev);
 
 	return 0;
+out:
+	kfree(pair);
+
+	return ret;
 }
 
 static int fsl_asrc_close(struct inode *inode, struct file *file)
-- 
1.7.9.5

